### PR TITLE
CloudWatch Logging and Backoff/Retry bug fixes

### LINKED ISSUE #135: DUM Client script does not respect configured logging level after a transfer failure/retry
### Checked for duplicates

No - I haven't checked

### 🐛 Describe the bug

Users of DUM have reported that when a sporadic upload failure occurs, the logging level reverts to DEBUG, providing far more console output than it should.

### 🕵️ Expected behavior

The DUM client should always respect the logging level provided either on the command line, or configured within the INI config.

### 📜 To Reproduce

Run DUM client and simulate a transfer failure

### 🖥 Environment Info

_No response_

### 📚 Version of Software Used

1.2.0

### 🩺 Test Data / Additional context

_No response_

### 🦄 Related requirements

🦄 #xyz

### ⚙️ Engineering Details

_No response_

### 🎉 Integration & Test

_No response_

### LINKED ISSUE #136: Backoff/Retry logic not firing for certain error codes
### Checked for duplicates

No - I haven't checked

### 🐛 Describe the bug

The backoff/retry decorator utilized with HTTP transfers in the DUM client is currently only configured to automatically retry when the HTTP failure code is 500 or above. Certain 4xx failures may also need to be added to retry other sporadic transfer errors.

### 🕵️ Expected behavior

The Backoff/retry giveup function should look for 4xx errors such as 400.

### 📜 To Reproduce

Run the DUM client against and endpoint that simulates or produces 4xx errors

### 🖥 Environment Info

- Version of this software [e.g. vX.Y.Z]
- Operating System: [e.g. MacOSX with Docker Desktop vX.Y]
...

### 📚 Version of Software Used

1.2.0

### 🩺 Test Data / Additional context

_No response_

### 🦄 Related requirements

🦄 #xyz

### ⚙️ Engineering Details

_No response_

### 🎉 Integration & Test

_No response_

### PR DESCRIPTION
**Pull request by collinss-jpl**
## 🗒️ Summary
This branch makes a number of modifications to how logs are submitted to CloudWatch to fix the following issues:

- Log level reverts to DEBUG whenever sending logs to CloudWatch fails for any reason
- The "giveup" function used with the backoff/retry decorator(s) does not properly handle all types of recoverable HTTP errors

The logging that occurs when a CloudWatch Logs send fails now utilizes a specialized "console-only" logger, rather than using the default "root" logger. Use of the "root" logger seemed to be the trigger that reset the global logging level to DEBUG. The `send_logs_to_cloud_watch` function is now properly wrapped with a backoff/retry decorator to further minimize any sporadic failures that might occur during regular flushing of logs to AWS.

The "giveup" function used with all backoff decorators now looks for specific HTTP codes that indicate a recoverable error (temporary server side unavailability or request throttling from AWS). This should also help to minimize the number of failed file uploads during regular usage of DUM.

## ⚙️ Test Data and/or Report
Unit tests for the logging utilities have been updated to account for changes made, and a new test has been added for usage of the backoff/retry decorator when submitting logs to CloudWatch.

This branch has also been deployed and tested on MCP dev as a manual integration test.

## ♻️ Related Issues

Fixes #135 
Fixes #136

### CODE REVIEW

File: src/pds/ingress/util/backoff_util.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    Determines if the HTTP return code associated with a requests exception
+    corresponds to a fatal error or not. If the error is of a transient nature,
+    this function will return False, indicating to the backoff decorator that
+    the reqeust should be retried. Otherwise, a return value of True will
+    cause any backoff/reties to be abandoned.
+    """
+    if err.response is not None:
+        # HTTP codes indicating a transient error (including throttling) which
+        # are worth retrying after a backoff
+        transient_codes = [408, 425, 429, 500, 502, 503, 504, 509]
```
Comment (nutjob4life) [inline src/pds/ingress/util/backoff_util.py:24]: I wonder if it'd be helpful to use the constants in [http.HTTPStatus](https://docs.python.org/3/library/http.html) ~as although I have memorized every one of the HTTP codes~ okay I have no idea what some of these codes are! 425 is "too early"? What‽

**Review by nutjob4life — APPROVED** [review]
Approved, but think about `http.HTTPStatus`

File: src/pds/ingress/util/backoff_util.py (hunk 1 above)
Comment (collinss-jpl) [reply src/pds/ingress/util/backoff_util.py]: Nice suggestion, I have added use of the enum everywhere I was using magic numbers for HTTP codes

**Review by nutjob4life — APPROVED** [review]
🎉

**Pull request merged**